### PR TITLE
create admin-page

### PR DESCRIPTION
**Pull request by kouch-chivorn**

### CODE REVIEW

File: app/controllers/admins/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Admins::RegistrationsController < ApplicationController
+  layout "admins/application"
+
+  def new
+  end
+
+  def create
+    redirect_to admins_root_path
+  end
+end
```
Comment (HorNorin) [inline app/controllers/admins/registrations_controller.rb]: new line

File: app/controllers/admins/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  layout "admins/application"
+
+  def index
+    @users = User.paginate page: params[:page], per_page: 10
+  end
+
+  def show
+    @user = User.find id: params[:id]
+  end
+end
```
Comment (HorNorin) [inline app/controllers/admins/users_controller.rb]: new line

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 class UsersController < ApplicationController
+  before_action :authenticate_user!, except: :new
   def show
     @user = User.find_by id: params[:id]
       redirect_to root_path,
```
Comment (HorNorin) [inline app/controllers/users_controller.rb]: indent

File: app/controllers/admins/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+class Admins::SessionsController < Devise::SessionsController
+  layout "admins/application"
+  def new
+    super
+  end
+
+  def create
+    super
+  end
+# before_filter :configure_sign_in_params, only: [:create]
```
Comment (chuanhtuan) [inline app/controllers/admins/sessions_controller.rb]: remove all remark code

File: app/controllers/admins/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class Admins::UsersController < ApplicationController
+  layout "admins/application"
+
+  def index
+    @users = User.paginate page: params[:page], per_page: 10
```
Comment (chuanhtuan) [inline app/controllers/admins/users_controller.rb]: plz use gem Setting

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -1,7 +1,9 @@
 class UsersController < ApplicationController
+  before_action :authenticate_user!, except: :new
+  
   def show
     @user = User.find_by id: params[:id]
-      redirect_to root_path, 
-      alert: "Not found user with #{params[:id]}" unless @user
+    redirect_to root_path, 
+    alert: "Not found user with #{params[:id]}" unless @user
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: indent

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -1,7 +1,9 @@
 class UsersController < ApplicationController
+  before_action :authenticate_user!, except: :new
+  
   def show
     @user = User.find_by id: params[:id]
-      redirect_to root_path, 
-      alert: "Not found user with #{params[:id]}" unless @user
+    redirect_to root_path,
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: no need

File: app/helpers/application_helper.rb [hunk 8]
```diff
@@ -1,2 +1,13 @@
 module ApplicationHelper
+  def resource_name
```
Comment (chuanhtuan) [inline app/helpers/application_helper.rb]: ???

File: app/helpers/application_helper.rb [hunk 9]
```diff
@@ -1,2 +1,13 @@
...
+    :user
+  end
+
+  def resource
```
Comment (chuanhtuan) [inline app/helpers/application_helper.rb]: ???

File: app/controllers/admins/users_controller.rb (hunk 5 above)
Comment (kouch-chivorn) [inline app/controllers/admins/users_controller.rb]: OK, I will use gem Setting in the new pull request after the admin sign-in and sign-out page

File: app/controllers/admins/registrations_controller.rb (hunk 1 above)
Comment (kouch-chivorn) [inline app/controllers/admins/registrations_controller.rb]: Done

File: app/controllers/admins/users_controller.rb (hunk 2 above)
Comment (kouch-chivorn) [inline app/controllers/admins/users_controller.rb]: Done

File: app/helpers/application_helper.rb (hunk 8 above)
Comment (kouch-chivorn) [inline app/helpers/application_helper.rb]: Those helper is for devise mapping, It does not work if I don't use it.

File: app/helpers/application_helper.rb (hunk 9 above)
Comment (kouch-chivorn) [inline app/helpers/application_helper.rb]: Those helpers are for devise mapping, It does not work if I don't use it.

**Pull request merged**